tests/PHPStan/Rules/Methods/data/bug-7519.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace Bug7519;
+
+use function PHPStan\Testing\assertType;
+use FilterIterator;
+
+/**
+ * @template TKey
+ * @template T
+ * @template TIterator as Iterator<TKey, T>
+ *
+ * @extends FilterIterator<TKey, T, TIterator>
+ */
+class A extends FilterIterator
+{
+	public function accept(): bool
+	{
+		return true;
+	}
+
+	public function key()
+	{
+		$key = parent::key();
+
+		assertType('TKey (class A, argument)', $key);
```

Comment:
This assertType isn't going to be checked. Because you're not referencing this file from NodeScopeResolverTest.

Reply:
sorry, shouldn't do this things in a hurry. pushed a fix.